ngraph_bridge/ie_executable.cc

```diff
@@ -141,5 +162,37 @@ bool IE_Executable::call(const vector<shared_ptr<runtime::Tensor>>& outputs,
   m_infer_req.Infer();
   return true;
 }
+
+bool IE_Executable::call_trivial(
+    const vector<shared_ptr<runtime::Tensor>>& outputs,
+    const vector<shared_ptr<runtime::Tensor>>& inputs) {
+  // outputs are in the same order as results
+  auto results = m_trivial_fn->get_results();
+  for (int i = 0; i < outputs.size(); i++) {
+    auto parent = results[i]->input_value(0).get_node_shared_ptr();
+    if (ngraph::is_type<opset::Parameter>(parent)) {
+      auto param = ngraph::as_type_ptr<opset::Parameter>(parent);
+      auto index = m_trivial_fn->get_parameter_index(param);
+      if (index < 0) {
+        THROW_IE_EXCEPTION << "Input parameter " << param->get_friendly_name()
+                           << " not found in trivial function";
+      }
+      auto size = inputs[index]->get_size_in_bytes();
+      unsigned char* buf_ptr = new unsigned char[size];
+      inputs[index]->read(buf_ptr, size);
+      outputs[0]->write(buf_ptr, size);
+      delete buf_ptr;
+    } else if (ngraph::is_type<opset::Constant>(parent)) {
+      auto constant = ngraph::as_type_ptr<opset::Constant>(parent);
+      outputs[i]->write(constant->get_data_ptr(),
+                        shape_size(constant->get_shape()) *
+                            constant->get_element_type().size());
+    } else {
+      THROW_IE_EXCEPTION << "Expected constant or parameter feeding to a "
```

Comment:
This will not happen, right? The call_trivial will only be called if the input is a constant or a param to a result.

Reply:
Yes, it won't happen – it's just overly defensive programming.